docs/specification/draft/basic/authentication.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Authentication
+type: docs
+weight: 15
+---
+
+{{< callout type="info" >}}
+**Protocol Revision**: {{< param protocolRevision >}}
+{{< /callout >}}
+
+## Overview
+
+The Model Context Protocol currently supports authentication through OAuth 2.0. Authentication is only available when using HTTP transport. Authentication occurs at the transport level. This specification defines how clients authenticate with MCP servers using standard OAuth 2.0 flows.
+
+## Authentication Flow
+
+MCP servers **MAY** require authentication for any request. When authentication is required and has not been provided, servers **MUST** respond with *HTTP 401 Unauthorized*. Clients **MUST** then initiate the OAuth 2.0 authentication flow.
+
+The basic authentication flow is as follows:
+
+```mermaid
+sequenceDiagram
+    participant C as Client
+    participant M as MCP Server
+    participant A as OAuth Server
+
+    C->>M: Any MCP Request
+    M->>C: HTTP 401 Unauthorized
+    C->>M: GET /authorize?redirect_uri=http://localhost:1234/callback
+    Note over M: Validate redirect_uri
+    M->>C: Redirect to OAuth Server
+    C->>A: OAuth Authorization Request
+    A->>M: Authorization Code (via redirect)
+    M->>A: Exchange Code for Tokens
+    A->>M: Access Token + Refresh Token
+    M->>C: Redirect to validated redirect_uri with session token
+    C->>M: Subsequent requests with session token
+    Note over C,M: Begin standard MCP message exchange
+```
+
+### Authorization Endpoints
+
+MCP servers **MUST** expose the following OAuth-related endpoints:
+
+1. `/authorize` - Initial authorization endpoint
+2. `/token` - Token exchange endpoint
+
+These endpoints are relative to the MCP server's base URL. For example, if the MCP server is at `https://example.com/mcp/`, the authorize endpoint would be `https://example.com/mcp/authorize`.
```

Comment:
What is written here seems fine, but i am not aware that there is already a strict definition of an MCP server url structure. I don't see URL anywhere mentioned in the spec website. Is this actually defined yet?

As an example when implementing a client using the SSE client library, my impression was I would need to give a base url to hand out was directly the endpoint e.g.  `https://example.com/some-path/see` and everything in relation is undefined (e.g. the post endpoint is only discovered based on the response, or by convention, but not actually part of the spec)

Reply:
You're correct about the current state. It may be the case that we want to make this stricter/more standardized going forward, though.